monkey/monkey_island/cc/ui/src/components/report-components/RansomwareReport.js

```diff
@@ -14,17 +14,62 @@ class RansomwareReport extends React.Component {
           <p>
             This report shows information about the ransomware simulation run by Infection Monkey.
           </p>
+          {this.getExploitationStats()}
         </div>
     )
   }
 
+  getExploitationStats() {
+    let num_scanned = this.props.report.propagation_stats.num_scanned_nodes;
+    let num_exploited = this.props.report.propagation_stats.num_exploited_nodes;
+
+    return (
+      <div>
+        <p>
+          The Monkey discovered <span className='badge badge-warning'>{num_scanned}</span> machines
+          and successfully breached <span className='badge badge-danger'>{num_exploited}</span> of them.
+        </p>
+        {this.getExploitationStatsPerExploit()}
+      </div>
+    )
+  }
+
+  getExploitationStatsPerExploit() {
+    let exploit_counts = this.props.report.propagation_stats.count_per_exploit;
+
+    let exploitation_details = [];
+
+    for (let exploit in exploit_counts) {
+      let count = exploit_counts[exploit];
+      if (count === 1) {
+        exploitation_details.push(
+          <div>
+            <span className='badge badge-danger'>{count}</span> machine was exploited by
+            the <span className='badge badge-danger'>{exploit}</span>.
+          </div>
+        );
+      }
+      else {
+        exploitation_details.push(
+          <div>
+            <span className='badge badge-danger'>{count}</span> machines were exploited by
+            the <span className='badge badge-danger'>{exploit}</span>.
+          </div>
+        );
+      }
+    }
```

Comment:
There's a lot of duplication here. I'd suggest something like
```suggestion
      let conjugated_machine_was = ""
      if (count === 1) {
        conjugated_machine_was = "machine_was"
      }
      else {
        conjugated_machine_was = "machines were"
      }
      
      exploitation_details.push(
        <div>
          <span className='badge badge-danger'>{count}</span>
          {conjugated_machine_was} exploited by the
          <span className='badge badge-danger'>{exploit}</span>.
        </div>
      );
    }
```

Reply:
Even better than this, we can look at using [Pluralize](https://github.com/plurals/pluralize) to do the conjugation and get rid of the if altogether.